backend/siarnaq/bracket/challonge.py

```diff
@@ -126,21 +126,118 @@ def get_tournament_data(tournament: Tournament, *, is_private: bool):
 
     r = requests.get(url, headers=_headers)
     r.raise_for_status()
-    return r.json()
+    data = r.json()
 
+    # Handle bracket-reset finals by internally splitting it up into two rounds,
+    # which Challonge does not do by default.
+    from siarnaq.api.episodes.models import TournamentStyle
+
+    if tournament.style == TournamentStyle.DOUBLE_ELIMINATION:
+        # Find the very last match (by play order),
+        # which is the second match of finals
+        # in case of bracket reset
+        matches = [item for item in data["included"] if item["type"] == "match"]
+        match_last = max(
+            matches, key=lambda match: match["attributes"]["suggestedPlayOrder"]
+        )
+        # Give it its own round
+        match_last["attributes"]["round"] += 1
+
+    return data
 
-def get_round_indexes(tournament: Tournament, *, is_private: bool):
+
+# TODO should this be private-only, and thus should I drop the is_private flag?
```

Comment:
@j-mao opinions?
Should we always get private data (for this method), thanks to the assumption that private and public have same structure etc?

gosh that seems odd

Reply:
Same as below, probably keep this kwarg, since it changes which bracket gets queried (even if currently useless).